src/lib/util/misc.c

```diff
@@ -1185,3 +1187,40 @@ int fr_digest_cmp(uint8_t const *a, uint8_t const *b, size_t length)
 
 	return result;		/* 0 is OK, !0 is !OK, just like memcmp */
 }
+
+/** Create an empty file. (It will overwrite an existing file).
+ *
+ * @param filename path to file.
+ * @param mode Specifies the file mode bits be applied.
+ */
+int fr_file_touch(char const *filename, mode_t mode) {
+	int fd;
+
+	fd = open(filename, O_WRONLY | O_CREAT, mode);
+	if (fd < 0) {
+		fr_strerror_printf("fr_file_touch() Failed creating file \"%s\": %s", filename, fr_syserror(errno));
+		return -1;
+	}
+
+	close(fd);
+
+	return 0;
+}
+
+/**  Remove a name from the filesystem.
+ */
+int fr_file_unlink(char const *filename) {
+	struct stat path_stat;
+
+	if (stat(filename, &path_stat) < 0) {
+		fr_strerror_printf("fr_file_unlink() Problems with stat(\"%s\")='%s'", filename, fr_syserror(errno));
+		return -1;		
+	}
+
+	if (!S_ISREG(path_stat.st_mode)) {
+		fr_strerror_printf("fr_file_unlink() only remove regular file: \"%s\": %s", filename, fr_syserror(errno));
+		return -1;
+	}
+
+	return remove(filename);
```

Comment:
Need to check the return code here and call fr_strerro_printf with fr_syserror.  The reason why we use fr_syserror is it's thread safe, and it prints the name of the error e.g. EINVAL or ENOENT in addition to the syserror text.

Reply:
This should also be ``unlink`` which is the POSIX function.